src/interfaces/pjson.ts

```diff
@@ -43,6 +43,8 @@ export namespace PJSON {
           hidden?: boolean;
         };
       };
+      helpOverride?: string[];
```

Comment:
I'm not sure that `helpOverride` and `versionOverride` are the best words here. We're not really overriding any behavior but instead adding to existing behavior.

Maybe it'd make more sense to make this `helpFlags` and `versionFlags`? 

As I'm thinking about it now, how useful is it to provide this functionality? Are users really going to use anything other than `-h` and `-v` for help and version? So perhaps it'd be better to have a couple of boolean properties that toggle on the short flags, e.g. `enableHelpShortFlag` and `enableVersionShortFlag`

I'm not sure what the right answer, I'm just thinking out loud here

Reply:
The name thing bothered me too. I'll think about new names.

For sfdx version we have `-v`, `--version` and `version`, so we have opened the door for flag names that are outside the short names. Core also supports the notion of command patterns like `<cli> --help` or `<cli> help` or `<cli> <cmd> [--help|help]` or `<cli> [--help|help] <cmd>`. Having a list of additional flags allows one to specify any name for "help."

IMO, I see us as the only ones going to use this capability, others will probably adapt to core's defaults.

Reply:
`help` and `version` are commands though, not flags. The `help` command is provided by [plugin-help](https://github.com/oclif/plugin-help) and the `--help` flag is provided by core. So if we're going to offer a `version` command I think we ought to make a plugin-version to be consistent with the existing pattern

I'm okay with providing the ability to configure any flag name but I agree that we might be the only ones who actually use it.

Reply:
These flags are intercepted in main.ts, with each having its own block [help](https://github.com/oclif/core/blob/phale/W-9612538/src/main.ts#L53) and [version](https://github.com/oclif/core/blob/phale/W-9612538/src/main.ts#L47). When a version flag or version command is detected it delegates to `config.userAgent`. When running `sf help deploy`, that is trapped by the block at line 53 and then delegated to a Help class. The only time I see main.ts delegate to the command, is when topic is used w/o a help flag.

Reply:
@peternhale I'm not sure that's correct. The debug logs indicate that `sf help deploy` is delegated to plugin-help:

```
  config runCommand help [ 'deploy' ] +1ms
  config:@oclif/plugin-help (require) /Users/mdonnalley/.nvm/versions/node/v14.15.4/lib/node_modules/@salesforce/cli/node_modules/@oclif/plugin-help/lib/commands/help.js +5ms
/Users/mdonnalley/.nvm/versions/node/v14.15.4/lib/node_modules/@salesforce/cli/node_modules/@sf/telemetry/lib/hooks/telemetryPrerun.js +0ms
  sf:help init version: @oclif/command@1.8.0 argv: [ 'deploy' ] +0ms
```

I don't think any additional works needs to be done in this PR. My point is just that if we're going to add a `version` command, it would be more consistent to place the functionality in a plugin.

Reply:
@mdonnalley I will compare current behavior wilh my changes to see if I have introduced an unwanted side ef
fect